pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -152,26 +153,33 @@ type ruleCache struct {
 	podUpdates <-chan v1beta1.PodReference
 }
 
-func (c *ruleCache) getNetworkPolicies(namespace string) []v1beta1.NetworkPolicy {
-	ret := []v1beta1.NetworkPolicy{}
+func (c *ruleCache) getNetworkPolicies(npFilter querier.NetworkPolicyQueryFilter) []v1beta1.NetworkPolicy {
+	var ret []v1beta1.NetworkPolicy
 	c.policyMapLock.RLock()
 	defer c.policyMapLock.RUnlock()
-	for uid, np := range c.policyMap {
-		if namespace == "" || np.Namespace == namespace {
+	for uid, npr := range c.policyMap {
+		if c.networkPolicyMatchFilter(npFilter, npr) {
 			ret = append(ret, *c.buildNetworkPolicyFromRules(uid))
 		}
 	}
 	return ret
 }
 
+// If this npr(Network Policy Reference) can match the npFilter(Network Policy Filter)
+func (c *ruleCache) networkPolicyMatchFilter(npFilter querier.NetworkPolicyQueryFilter, npr v1beta1.NetworkPolicyReference) bool {
+	return (npFilter.Name == "" || npFilter.Name == npr.Name) &&
+		(npFilter.Namespace == "" || npFilter.Namespace == npr.Namespace) &&
+		(npFilter.SourceType == "" || string(npFilter.SourceType) == string(npr.Type))
```

Comment:
I think `npFilter.SourceType` and `npr.Type` can be compared directly without being converted to string first.

Reply:
Thx!